### DISCUSSION TITLE
Request for Metadata Overhaul and Community Push

### DISCUSSION THREAD

**Discussion by ShiningMinions** [sunflower-land/sunflower-land#1539] [General]
### TLDR Points;
- Currently, SFL's item metadata exists and is good, but there are not many data points. It would be really nice if SFL's item metadata was greatly overhauled, providing a large set of useful datapoints.
- These more-specific datapoints would be extremely useful in external indexing, filtering, searching, tooling and calculations (including sfl.tools).
- I also think these more-specific datapoints should reside in SFL's metadata repo, so that the community doesn't need to keep a track of their own independent indexes and we can all share a common base.
- This would also be very useful for openSea buyers/sellers.

# Proposal
I would like to propose an overhaul of SFL's current metadata + implement a new metadata-adding workflow (I would be more than willing to lead this charge). If this proposal gains traction and acceptance, I will create a GitHub issue going over the technical specifics of the new metadata generation script + workflow. What I would need help with from the community is to add some of these missing dataPoints across all SFL items after the new workflow is implemented. My vision is that adding a new item's metadata would be extremely easy and streamlined, and the reoccurring problem of missing metadata would stop happening. This would also be a great opportunity for new contributors to contribute to SFL :eyes: .

## Examples of Missing Metadata I Would Like to See:

Let's use this beetroot on openSea as an example:

![beetroot-opensea](https://user-images.githubusercontent.com/103600068/196443229-8d19d7d0-56b7-454e-a019-1657b989e5df.png)

Here are some additional metaData I would like to see on this record:
- Crop Growth Time
- In-game Shop Sell Price
- Is Withdrawable from Farm? ("Tradable" field is confusing)
- Is Tradable at Goblin Trader?
- A link to the seed which grows it
- A link to the type of food you need in order to unlock this crop

This is just an example, specifically for a crop. But for other things like NFTs, I would like to see things like:
- Mint cooldown time
- Max supply (not in markdown, but in unique field)
- Links to the recipe items
- etc...

**Comment by adamhannigan** [comment]
@ShiningMinions Agreed.

There are lots of characteristics that would be useful in these cases.

What do you feel the best format for capturing these would be? It would also be nice to think how this fits into the processes - i.e. include in typescript/CI and throw errors if developers forget to add it

**Reply by ShiningMinions** [reply]
> What do you feel the best format for capturing these would be? It would also be nice to think how this fits into the processes - i.e. include in typescript/CI and throw errors if developers forget to add it

@adamhannigan There are a few things I'm thinking about... perhaps I should open a github chore request issue to communicate some more technical specifics. But I'm thinking something like this:

1. First, having some nice script that make adding metadata very easy:
```bash

$ ts-node addItem.ts

> Enter the blockChain ID for this item:
1104

> Select the type of this item:
[ ] Seed
[✓] Crop
[ ] Tool
[ ] (etc...)
[ ] Add New Category

etc...
```
2. The result of this script is that it will generate a markdown file with `front-matter`. I think front-matter would be a solid choice instead of hard-parsing ambiguous markdown lines when come time to generate the .json files. For instance, beetroot markdown file could look like:
```markdown
---
id: 205
name: Beetroot
type: Crop
isWithdrawable: true
isGoblinTradable: true
maxSupply: unlimited
cropDetails:
    shopSellPrice: 0.035
    seedId: 106
etc...
---

(This is where the OpenSea description would go, right beneath the front-matter). i.e:

Apparently, they’re an aphrodisiac...
```

You could of course manually edit these markdown files, but the script would just make it much easier.

3. To ensure developers add item metadata, perhaps it would be interesting to do a husky pre-commit hook here. This pre-commit script could look at the KNOWN_IDs file, and verify that there exists a markdown file for every ID listed. This script could also run in CI pipeline perhaps.

4. Speaking of CI, I'm thinking the SFL git repo should stop tracking generated .json files. These .json files should be .gitignore'd, and instead be automatically generated from the markdown files in a post-release CI step.

What do you think?

**Comment by ShiningMinions** [comment]
@adamhannigan just wanted to bump this discussion - please see previous response. Would this be an acceptable effort for me to start tackling?
